lib/Sema/TypeCheckType.cpp

```diff
@@ -2705,7 +2705,8 @@ Type TypeResolver::resolveASTFunctionType(
   switch (representation) {
   case AnyFunctionType::Representation::Block:
   case AnyFunctionType::Representation::CFunctionPointer:
-    if (!fnTy->isRepresentableIn(ForeignLanguage::ObjectiveC, DC)) {
+    if (!fnTy->hasError() &&
+        !fnTy->isRepresentableIn(ForeignLanguage::ObjectiveC, DC)) {
```

Comment:
Side note: should `isRepresentableIn` really be allowed to succeed on or even accept malformed types?

Reply:
I'm not sure, but just by a quick look in `getForeignRepresentable` I think it will always fail (if you mean fail as return false not assert) on malformed types. But maybe it could be a fast path early return?

Reply:
We should also aim to propagate ErrorType early. What happens if you lift this up so it never sees FunctionTypes with ErrorTypes in them?

Reply:
Do you mean in `getForeignRepresentable`? I did a test yesterday return early there for error types and all tests passed(only `/tests` didn't run validation). So nothing really happened :)

Reply:
I think what Robert meant is to see what happens if we return on `fnTy->hasError` before we switch on the representation.

Reply:
Ahh right, I just adjusted :)

Reply:
cc @CodaFi